src/llvm-version.h

```diff
@@ -2,53 +2,34 @@
 
 #include <llvm/Config/llvm-config.h>
 
-#if defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 4 && LLVM_VERSION_MINOR >= 0
-#define LLVM40 1
-#endif
+// The LLVM version used, JL_LLVM_VERSION, is represented as a 5-digit integer
+// of the form ABBCC, where A is the major version, B is minor, and C is patch.
+// So for example, LLVM 3.7.0 is 30700.
+#define JL_LLVM_VERSION (LLVM_VERSION_MAJOR * 10000 + LLVM_VERSION_MINOR * 100 \
+                    + LLVM_VERSION_PATCH)
 
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 9)
-#define LLVM39 1
+#if JL_LLVM_VERSION < 30300
```

Comment:
So dropping (source code) support for 3.4-3.6? (after effectively dropped it for a long time).